Improve schema explanations - Closes #1236 (#1320)
LiskHQ/lisk-docs: modules/ROOT/pages/advanced-explanations/schemas.adoc

NOTE: If the data type of a property is either an `object` or an `array`, the `type` property must be used instead of `dataType`.

.Example of an account schema of a module
[source,js]
----
export const lnsAccountPropsSchema = {
	$id: 'lisk/lns/lnsAccount',
	type: 'object',
	required: ['ownNodes', 'reverseLookup'],
	properties: {
		reverseLookup: {
			dataType: 'bytes',
			fieldNumber: 1,
		},
		ownNodes: {
			type: 'array',
			fieldNumber: 2,
			items: {
				dataType: 'bytes',
			},
		},
	},
	default: {
		ownNodes: [],
		reverseLookup: EMPTY_BUFFER,
	},
};
----

=== $id

Unique identifier of the schema throughout the system.

The $id property is straightly inherited from the JSON-schema.
You can read more about the `id` property on {url_json_schema_id}[^].

In general:

> * Use unique ids across the system

Author: ```suggestion
* Use unique IDs across the system
```

* Use path like format (for easy readability not a requirement)

> * Use one fixed identifier for your app in each id to not mix any schema other registered schema.

Author: ```suggestion
* in order to ensure no mixing of any schema with other registered schemas occurs, use one fixed identifier for your app in each ID
* ```

== Converting between formats

Account addresses are stored as binary data in the database.
But in other parts of the application, the data is expected as a hexadecimal string, which is more human-readable.

> On the other hand, the node API returns the data like blocks and transactions always in hex string, but to decode transactions or blocks, they need to be in `Buffer` format.

Reviewer: ```suggestion
On the other hand, the node API always returns the data such as blocks and transactions in hex string. However, please be aware that to decode transactions or blocks, they need to be in `Buffer` format.
```
